hardware_interface/src/component_parser.cpp

```diff
@@ -200,6 +263,52 @@ ComponentInfo parse_component_from_xml(const tinyxml2::XMLElement * component_it
   return component;
 }
 
+/// Search XML snippet from URDF for information about a complex component.
+/**
+ * A complex component can have a non-double data type specified on its interfaces,
+ *  and the interface may be an array of a fixed size of the data type.
+ *
+ * \param[in] component_it pointer to the iterator where component
+ * info should befound
+ * \throws std::runtime_error if a required component attribute or tag is not found.
+ */
+ComponentInfo parse_complex_component_from_xml(
+  const tinyxml2::XMLElement * component_it)
+{
+  ComponentInfo component;
+
+  // Find name, type and class of a component
+  component.type = component_it->Name();
+  component.name = get_attribute_value(component_it, kNameAttribute, component.type);
+
+  // Parse all command interfaces
+  const auto * command_interfaces_it = component_it->FirstChildElement(kCommandInterfaceTag);
+  while (command_interfaces_it) {
+    component.command_interfaces.push_back(parse_interfaces_from_xml(command_interfaces_it));
+    component.command_interfaces.back().data_type =
+      parse_data_type_attribute(command_interfaces_it);
```

Comment:
Parsing of `size` and `data_type` should be probably called in the above method `parse_interfaces_from_xml` to simplify this part of the code. This could be also done in a follow-up PR.